titan-server/src/teams/routes.rs

```diff
@@ -188,18 +164,18 @@ pub fn get_parent_role(
     titan_db: TitanPrimary,
     wcf_db: UnksoMainForums,
     app_config: State<config::AppConfig>
-) -> Json<Option<models::OrganizationRoleWithAssoc>> {
+) -> ApiResponse<Option<models::OrganizationRoleWithAssoc>> {
     let titan_db_ref = &*titan_db;
-    let parent_role_res = teams::roles::find_parent_role(
-        role_id, false, titan_db_ref).unwrap();
-    match parent_role_res {
-        Some(parent_role) => {
-            let parent_role_assoc = teams::roles::map_role_assoc(
-                &parent_role, titan_db_ref, &*wcf_db, &app_config).unwrap();
-            Json(Some(parent_role_assoc))
-        },
-        None => Json(None)
-    }
+    let res = teams::roles::find_parent_role(role_id, false, titan_db_ref)
+        .and_then(|parent_role| {
+            match parent_role {
+                Some(role) =>  teams::roles::map_role_assoc(
+                    &role, titan_db_ref, &*wcf_db, &app_config)
+                    .map(|role_assoc| Some(role_assoc)),
```

Comment:
The closure appears to be redundant. You should just be able to send `Some` through as an argument to `map`.

For example:

```rust
teams::roles::map_role_assoc(..).map(Some)
```